creusot/tests/should_succeed/red_black_tree.rs

```diff
@@ -574,7 +574,7 @@ where
         self.node.as_mut().unwrap().color = Black;
         proof_assert! { forall<h: Int> old_self.has_height(h) ==>
         (*self).has_height(h) || (*self).has_height(h+1) }
-        proof_assert! { (*self).has_mapping_model(@key /* dummy */); true }
+        ghost! { Self::has_mapping_model };
```

Comment:
From what I can tell, the reason you need to translate`FnDef` is that you have bare function calls in these ghost macros, I wonder if we could avoid that.

Reply:
So you don't think having these bare functions is a good thing?

As far as I'm concerned, I don't see why having to translate FnDef is a problem.

Reply:
If we had non-linear ghost code then it would be easy to handle since we could move out of self to provide the arguments to `has_mapping_model`.

Reply:
> As far as I'm concerned, I don't see why having to translate FnDef is a problem.

I have an issue with translating them to `()`, if we could translate them to function types I wouldn't really care.

Reply:
Well, the thing is `FnDef` types are really singletons. So it really makes sense to translate them to unit.

Reply:
Oh, true, perharps `FnPtr` could (one day) be translated as function types.

Reply:
Can you add a comment above that line in `ty.rs`?